handler/perm_handlers_test.go

```diff
@@ -0,0 +1,247 @@
+package handler
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"net/http/httptest"
+	"stlab.itechart-group.com/go/food_delivery/authorization_service/model"
+	"stlab.itechart-group.com/go/food_delivery/authorization_service/pkg/logging"
+	"stlab.itechart-group.com/go/food_delivery/authorization_service/service"
+	mock_service "stlab.itechart-group.com/go/food_delivery/authorization_service/service/mocks"
+	"testing"
+)
+
+func TestHandler_getPermsByRoleId(t *testing.T) {
+	type mockBehavior func(s *mock_service.MockRolePerm, id int)
+
+	testTable := []struct {
+		name                string
+		input               string
+		id                  int
+		mockBehavior        mockBehavior
+		expectedStatusCode  int    //expected code
+		expectedRequestBody string //expected response
+	}{
+		{
+			name:  "OK",
+			input: "1",
+			id:    1,
+			mockBehavior: func(s *mock_service.MockRolePerm, id int) {
+				s.EXPECT().GetPermsByRoleId(id).Return([]model.Permission{
+					{
+						ID:   1,
+						Name: "test",
+					},
+					{
+						ID:   2,
+						Name: "test2",
+					},
+				}, nil)
+			},
+			expectedStatusCode:  200,
+			expectedRequestBody: `{"Perms":[{"id":1,"name":"test"},{"id":2,"name":"test2"}]}`,
+		},
+		{
+			name:                "invalid request",
+			input:               "a",
+			mockBehavior:        func(s *mock_service.MockRolePerm, id int) {},
+			expectedStatusCode:  400,
+			expectedRequestBody: `{"message":"Invalid request"}`,
+		},
+		{
+			name:  "server error",
+			input: "1",
+			id:    1,
+			mockBehavior: func(s *mock_service.MockRolePerm, id int) {
+				s.EXPECT().GetPermsByRoleId(id).Return(nil, fmt.Errorf("server error"))
+			},
+			expectedStatusCode:  500,
+			expectedRequestBody: `{"message":"server error"}`,
+		},
+	}
+
+	for _, testCase := range testTable {
+		t.Run(testCase.name, func(t *testing.T) {
+			//Init dependencies
+			c := gomock.NewController(t)
+			defer c.Finish()
+			getPerm := mock_service.NewMockRolePerm(c)
+			testCase.mockBehavior(getPerm, testCase.id)
+			logger := logging.GetLogger()
+			services := &service.Service{RolePerm: getPerm}
+			handler := NewHandler(services, logger)
+
+			//Init server
+			r := handler.InitRoutes()
+
+			//Test request
+			w := httptest.NewRecorder()
```

Comment:
It is better to treat test code similar to production code and create helper functions that setup your fake server and teardown it when needed